src/canvas-view/canvas/canvas-component.js

```diff
@@ -0,0 +1,411 @@
+          ctrl.chart.resetValidNodesAndConnectors();
+        };
+
+        //
+        // Handle connectionMouseOver on an connection.
+        //
+        ctrl.connectionMouseOver = function (evt, connection) {
+          if (!ctrl.draggingConnection && !ctrl.readOnly) {  // Only allow 'connection mouse over' when not dragging out a connection.
+            ctrl.mouseOverConnection = connection;
+          }
+        };
+
+        //
+        // Handle connectionMouseLeave on an connection.
+        //
+        ctrl.connectionMouseLeave = function () {
+          ctrl.mouseOverConnection = null;
+        };
+
+        //
+        // Handle mousedown on a connection.
+        //
+        ctrl.connectionMouseDown = function (evt, connection) {
+          var chart = ctrl.chart;
+          if (!ctrl.readOnly) {
+            chart.handleConnectionMouseDown(connection, evt.ctrlKey);
+          }
+          // Don't let the chart handle the mouse down.
+          evt.stopPropagation();
+          evt.preventDefault();
+        };
+
+        //
+        // Handle connectorMouseOver on an connector.
+        //
+        ctrl.connectorMouseOver = function (evt, node, connector) {
+          if (!ctrl.readOnly) {
+            ctrl.mouseOverConnector = connector;
+          }
+        };
+
+        //
+        // Handle connectorMouseLeave on an connector.
+        //
+        ctrl.connectorMouseLeave = function () {
+          ctrl.mouseOverConnector = null;
+        };
+
+        //
+        // Handle mousedown on an input connector.
+        //
+        ctrl.connectorMouseDown = function (evt, node) {
+          if (ctrl.chart.inConnectingMode && node !== ctrl.connectingModeSourceNode) {
+            ctrl.chart.createNewConnection(ctrl.connectingModeOutputConnector, ctrl.mouseOverConnector);
+            ctrl.stopConnectingMode();
+          }
+        };
+
+        //
+        // zoom.
+        //
+        $scope.$on('zoomIn', function () {
+          ctrl.chart.zoom.in();
+        });
+
+        $scope.$on('zoomOut', function () {
+          ctrl.chart.zoom.out();
+        });
+
+        $scope.maxZoom = function () {
+          return (ctrl.chart.chartViewModel && ctrl.chart.chartViewModel.zoom) ? ctrl.chart.chartViewModel.zoom.isMax() : false;
+        };
+        $scope.minZoom = function () {
+          return (ctrl.chart.chartViewModel && ctrl.chart.chartViewModel.zoom) ? ctrl.chart.chartViewModel.zoom.isMin() : false;
+        };
+
+        ctrl.zoomLevel = function () {
+          return ctrl.chart.zoom.getLevel();
+        };
+
+        ctrl.$onInit = function () {
+          var backspaceKeyCode = 8;
+          var deleteKeyCode = 46;
+          var ctrlKeyCode = 17;
+          var ctrlDown = false;
+          var aKeyCode = 65;
+          var escKeyCode = 27;
+
+          $document.find('body').keydown(function (evt) {
+            if (evt.keyCode === ctrlKeyCode) {
+              ctrlDown = true;
```

Comment:
You can check evt.ctrlKey to determine if ctrl is pressed rather than keeping state.